### PR TITLE
Fixes #1691 Latest gocb breaks go get

### LINKED ISSUE #1691: Latest gocb breaks go get
Using go get to build Sync Gateway fails when picking up the latest gocb - there are changes to the gocb logger that break our current logging integration.  We should move to the latest gocb in the manifest and make the required fixes.

### PR DESCRIPTION
**Pull request by tleyden**

### CODE REVIEW

File: base/bucket_gocb.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ type CouchbaseBucketGoCB struct {
 
 type GoCBLogger struct{}
 
-func (l GoCBLogger) Output(s string) error {
-	LogTo("gocb", s)
+func (l GoCBLogger) Log(level gocbcore.LogLevel, offset int, format string, v ...interface{}) error {
```
Comment (adamcfraser) [inline base/bucket_gocb.go:64]: What do you think about using the gocbcore.LogLevel to switch to log.Warn or log.LogError?

Comment (tleyden) [inline base/bucket_gocb.go:64]: Good call!  Will make that change

**Comment by tleyden** [comment]
Made that change, rebuild in progress: http://drone.couchbasemobile.com/couchbase/sync_gateway/19

File: base/bucket_gocb.go [hunk 2]
```diff
@@ -61,8 +61,13 @@ type CouchbaseBucketGoCB struct {
...
+	switch level {
+	case gocbcore.LogError:
+		LogError(fmt.Errorf(format, v))
+	default:
```
Comment (adamcfraser) [inline base/bucket_gocb.go:70]: I was expecting to also use Warn for gocbcore.LogWarn - let me know if you left that out intentionally.

**Comment by tleyden** [comment]
> I was expecting to also use Warn for gocbcore.LogWarn - let me know if you left that out intentionally.

I added this.  

I guess originally I was trying to maximize the usage of `LogTo` in the sense that it preserves the tag.  I'm wondering if we should add `LogErrrorTo(tag, etc)` at some point.

**Comment by adamcfraser** [comment]
I don't know if we'd want to filter out based on tag for warnings/errors, but I agree that we're losing some contextual information when using Warn or Error - it would be useful to know that the Warning/Error is being raised by gocb.  I like the idea of LogErrorTo and LogWarnTo that would prepend the log message with the log key.

**Comment by tleyden** [comment]
Passing build: http://drone.couchbasemobile.com/couchbase/sync_gateway/25

**Pull request merged**